receiver/riakreceiver/scraper.go

```diff
@@ -0,0 +1,115 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package riakreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/riakreceiver"
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/riakreceiver/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/riakreceiver/internal/models"
+)
+
+var errClientNotInit = errors.New("client not initialized")
+
+// riakScraper handles scraping of Riak metrics
+type riakScraper struct {
+	logger   *zap.Logger
+	cfg      *Config
+	settings component.TelemetrySettings
+	client   client
+	mb       *metadata.MetricsBuilder
+}
+
+// newScraper creates a new scraper
+func newScraper(logger *zap.Logger, cfg *Config, settings component.TelemetrySettings) *riakScraper {
+	return &riakScraper{
+		logger:   logger,
+		cfg:      cfg,
+		settings: settings,
+		mb:       metadata.NewMetricsBuilder(metadata.DefaultMetricsSettings()),
+	}
+}
+
+// start starts the scraper by creating a new HTTP Client on the scraper
+func (r *riakScraper) start(ctx context.Context, host component.Host) (err error) {
+	r.client, err = newClient(r.cfg, host, r.settings, r.logger)
+	return
+}
+
+// scrape collects metrics from the Riak API
+func (r *riakScraper) scrape(ctx context.Context) (pdata.Metrics, error) {
+	metrics := pdata.NewMetrics()
+
+	// Validate we don't attempt to scrape without initializing the client
+	if r.client == nil {
+		return metrics, errors.New("client not initialized")
```

Comment:
Looks safe to just initialize here.

Reply:
@djaglowski Just a quick question about doing this, I noticed that alot of the other receivers check to see if the client is initialized but only returns an error. Since this is in the scrape function is already assuming that the client is initialized, would it be better to return an error if it is not?

Reply:
It's something that isn't entirely consistent across scrapers. I tend to favor the more robust approach, but calling `scraper` without first calling `start` is not a primary use case here. I'm fine with leaving it as is, and this can be a point of standardization later.